File: ios/engine/KMEI/KeymanEngine/Classes/Categories/UIButton+Helpers.m
//
//  UIButton+Helpers.m
//  KeymanEngine
//
//  Created by Serkan Kurt on 5/03/2014.
//  Copyright (c) 2017 SIL International. All rights reserved.
//

#import "UIButton+Helpers.h"
#import <QuartzCore/QuartzCore.h>

@implementation UIButton (Helpers)

- (void)setRoundedBorderWithRadius:(float)radius borderWidth:(float)borderWidth color:(UIColor*)color {
    CALayer *layer = [self layer];
    [layer setCornerRadius:radius];
    [layer setBorderWidth:borderWidth];
    [layer setBorderColor:[color CGColor]];
    [layer setMasksToBounds:YES];
}

@end

File: ios/engine/KMEI/KeymanEngine/Classes/Extension/UIButton+Helpers.swift
//
//  UIButton+Helpers.swift
//  KeymanEngine
//
//  Created by Gabriel Wong on 2017-10-16.
//  Copyright © 2017 SIL International. All rights reserved.
//

import UIKit

extension UIButton {
  func setRoundedBorder(withRadius radius: CGFloat, borderWidth: CGFloat, color: UIColor) {
    layer.cornerRadius = radius
    layer.borderWidth = borderWidth
    layer.borderColor = color.cgColor
    layer.masksToBounds = true
  }
}
